src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1326,10 +1326,11 @@ public override bool IsTupleCompatible(out int tupleCardinality)
 
                     do
                     {
+                        Debug.Assert(!typeToCheck.IsDefinition);
```

Comment:
>Debug.Assert(!typeToCheck.IsDefinition); [](start = 24, length = 40)

I don't think we can make this assumption. The purpose of the function is to verify it.